backend/model/jobPoster.js

```diff
@@ -116,6 +117,61 @@ class JobPoster extends Job {
     }
   }
 
+  /*
+	__process_event in the base Job class deals with listen for events on
+	current job instances. In order for a worker to start the bidding process,
+	we only care about `JobDescriptionPosted` if the client meets certainn run
+	time states. We override __precess event below to make that happen.
+	*/
+  static __process_event(name, instanceId, event) {
+    console.log("poster tryCatch ");
+    try {
+      //Getting confirmation of a job that this wallet just posted
+      if (
+        name === "JobDescriptionPosted" &&
+        event.returnValues.jobPoster === wallet.address &&
+        this.preConfirmedJobs.length > 0
+      ) {
+        //Find the job from preConfirmed
+        let job = this.preConfirmedJobs.shift();
+
+        job.transactions.push({ event: "postJobDescription" });
+
+        // Gather data about the job
+        job.jobData = event.returnValues;
+
+        // Hold the this job in the jump table
+        job.addToJump();
```

Comment:
Yep, so to understand this, we first need to look at the events() fuinction here: https://github.com/morphware/daemon/blob/1f99cc71736e384bb87b66bca82e651e3250f6de/backend/model/job.js#L123

What this does it listen to all events emmited from both the auctionFactory contract and the jobFactory contract. Once it listens to the event, it parse the event by sending it over to the __parseEvent function defined here

https://github.com/morphware/daemon/blob/1f99cc71736e384bb87b66bca82e651e3250f6de/backend/model/job.js#L145

However, since this bug was specific to the jobWorker. I overrided the __parseEvent function in the parent class (Job), by re-defining it in JobPoster here.

The code here will act on behalf of the JobPoster iff the event is  related to this wallet, is a JobDescriptionPosted event and the current preConfirmedJobs is not empty. Because then this is the event that validates that the job we posted has infact completed its transaction, and we can not move it from the preConfirmedJobs array into the jump table

Reply:
Hey @darshanraju thanks for the explanation. Super helpful. I am trying to build a mental model of the workflow. I think now i get that from the job poster side, it's a job goes from preConfirmedTable >> Jump table when the jobDecriptionPosted event was found. So the Jump table is equivalent to Confirmed jobs array?

Reply:
Yep thats right, the JumpTable stores confirmed jobs. And any API's that request any data around jobs, the daemon will respond with data from the jump tablbe